### PR TITLE
Fixed points real

### PR DESCRIPTION
**Pull request by Thanhson89**

### CODE REVIEW

File: pysmt/rewritings.py [hunk 1]
```diff
@@ -1223,6 +1224,194 @@ def __init__(self, environment=None):
...
+
+    def walk_sfxp_div(self, formula, args, **kwargs):
+        ty = self.env.stc.get_type(formula)
+        total_width = ty.total_width
+        frac_width = ty.frac_width
+        om = args[0]
+        rm = args[1]
+        left = args[2]
+        right = args[3]
+        result = self.mgr.Times(left,right)
```
Comment (keram88) [inline pysmt/rewritings.py]: Should be self.mgr.Div

File: pysmt/rewritings.py [hunk 2]
```diff
@@ -1223,6 +1224,194 @@ def __init__(self, environment=None):
...
+    def walk_ufxp_div(self, formula, args, **kwargs):
+        ty = self.env.stc.get_type(formula)
+        total_width = ty.total_width
+        frac_width = ty.frac_width
+        om = args[0]
+        rm = args[1]
+        left = args[2]
+        right = args[3]
+        result = self.mgr.Div(left,right)
+        return self.process_real_round(result,om,rm,0,total_width,frac_width)
```
Comment (keram88) [inline pysmt/rewritings.py:1359]: This needs to handle the case of a zero divisor:
self.mgr.Ite(self.mgr.Equals(right, Fraction(0)), Fraction(2**total_width-1), ...)
Likewise for sfxp_div

File: pysmt/rewritings.py [hunk 3]
```diff
@@ -1235,11 +1424,27 @@ def walk_ufxp_constant(self, formula, args, **kwargs):
...
-    def walk_bv_constant(self, formula, args, **kwargs):
-        return formula
+
+    def walk_ufxp_lt(self, formula, args, **kwargs):
+        return self.mgr.LT(args[0], args[1])
+
+    def walk_ufxp_le(self, formula, args, **kwargs):
+        return self.mgr.LE(args[0], args[1])
+
+    def walk_sfxp_lt(self, formula, args, **kwargs):
```
Comment (keram88) [inline pysmt/rewritings.py]: Should be `walk_sfxp_gt`.

File: pysmt/rewritings.py [hunk 4]
```diff
@@ -1235,11 +1424,27 @@ def walk_ufxp_constant(self, formula, args, **kwargs):
...
+        return self.mgr.LT(args[0], args[1])
```
Comment (keram88) [inline pysmt/rewritings.py]: Should be `self.mgr.GT`. Do the same for `walk_sfxp_le`->`walk_sfxp_ge` below.

**Review by keram88 — CHANGES_REQUESTED** [review]

File: pysmt/rewritings.py (hunk 2 above)
Comment (keram88) [reply pysmt/rewritings.py:1359]: Actually we need to change PySMT to make this work.

**Comment by Thanhson89** [comment]
Fixed, thanks

**Pull request merged**